core/src/components/modal/gestures/swipe-to-close.ts

```diff
@@ -35,34 +39,82 @@ export const createSwipeToCloseGesture = (
   };
 
   const onStart = () => {
+    /**
+     * If canDismiss is anything other than `true`
+     * then users should be able to swipe down
+     * until a threshold is hit. At that point,
+     * the card modal should not proceed any further.
+     * TODO (FW-937)
+     * Remove undefined check
+     */
+    canDismissBlocksGesture = el.canDismiss !== undefined && el.canDismiss !== true;
     animation.progressStart(true, (isOpen) ? 1 : 0);
   };
 
   const onMove = (detail: GestureDetail) => {
-    const step = clamp(0.0001, detail.deltaY / height, 0.9999);
-
-    animation.progressStep(step);
+    const step = detail.deltaY / height;
+
+    /**
+     * Check if user is swiping down and
+     * if we have a canDismiss value that
+     * should block the gesture from
+     * proceeding,
+     */
+    const isAttempingDismissWithCanDismiss = step >= 0 && canDismissBlocksGesture;
+
+    /**
+     * If we are blocking the gesture from dismissing,
+     * set the max step value so that the sheet cannot be
+     * completely hidden.
+     */
+    const maxStep = isAttempingDismissWithCanDismiss ? canDismissMaxStep : 0.9999;
+
+    /**
+     * If we are blocking the gesture from
+     * dismissing, calculate the spring modifier value
+     * this will be added to the starting breakpoint
+     * value to give the gesture a spring-like feeling.
+     * Note that the starting breakpoint is always 0,
+     * so we omit adding 0 to the result.
+     */
+    const processedStep = isAttempingDismissWithCanDismiss ? calculateSpringStep(step / maxStep) : step;
+
+    const clampedStep = clamp(0.0001, processedStep, maxStep);
+
+    animation.progressStep(clampedStep);
   };
 
   const onEnd = (detail: GestureDetail) => {
     const velocity = detail.velocityY;
+    const step = detail.deltaY / height;
+
+    const isAttempingDismissWithCanDismiss = step >= 0 && canDismissBlocksGesture;
+    const maxStep = isAttempingDismissWithCanDismiss ? canDismissMaxStep : 0.9999;
+
+    const processedStep = isAttempingDismissWithCanDismiss ? calculateSpringStep(step / maxStep) : step;
 
-    const step = clamp(0.0001, detail.deltaY / height, 0.9999);
+    const clampedStep = clamp(0.0001, processedStep, maxStep);
 
     const threshold = (detail.deltaY + velocity * 1000) / height;
 
-    const shouldComplete = threshold >= 0.5;
+    /**
+     * If canDismiss blocks
+     * the swipe gesture, then the
+     * animation can never complete until
+     * canDismiss is checked.
+     */
+    const shouldComplete = !isAttempingDismissWithCanDismiss && threshold >= 0.5;
```

Comment:
Since line 142 checks `isAttempingDismissWithCanDismiss` regardless of `shouldComplete`'s value, I think the `isAttempting` check here is redundant?

Reply:
`shouldComplete` also controls whether the card snaps to the beginning or closes completely: https://github.com/ionic-team/ionic-framework/pull/24920/files#diff-884014c450143e2903ace69c69f0b285e483f5886cc6d480775545350a351708R128

If `isAttempingDismissWithCanDismiss` is `true`, we want the card to snap back to the beginning even if `threshold >= 0.5`

Reply:
Gotcha, makes sense 👍